pkg/controller/instance/controller.go

```diff
@@ -135,6 +137,10 @@ func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) error {
 	// runtime object in it's fields.
 	rgRuntime, err := c.rgd.NewGraphRuntime(instance)
 	if err != nil {
+		// Mark graph resolution failure and update status before returning error
+		mark := NewConditionsMarkerFor(instance)
+		mark.GraphNotResolved("failed to create runtime resource graph definition: %v", err)
+		c.updateInstanceStatusOnError(ctx, instance)
 		return fmt.Errorf("failed to create runtime resource graph definition: %w", err)
 	}
 
```

Comment:
do we want to set GraphResolved right after this block?

Reply:
it happens in the reconciler.